src/monitor/tracepoint_monitor.cpp

```diff
@@ -39,9 +39,8 @@ TracepointMonitor::TracepointMonitor(trace::Trace& trace, Cpu cpu)
     for (const auto& event_name : config().tracepoint_events)
```

Comment:
This does not exactly work, since calling the initializer from config means that not all config values are set (possibly) and I can't create a complete TracepointEvent. (The error that will occur is "No value set")

I could however create invalid TracepointEvents in the initializer that can at least be iterated over and replaced with a valid one when needed.

Reply:
The issue is values taken from config()?

I think that it is not valid  to build the *Event classes in a way that they can only be correctly be constructed after we have a config(). This leads to many cases where the order in which things are constructed ends up with the *Event types blowing up in our faces, because some class containing a PerfEvent somewhere get constructed before the config is.

Please rewrite the code *Event code  so it does not read it directly from the config but has setters like set_foo, set_bar ...

Reply:
I'm tempted to either add a init_attrs_with_config() function, which sets all of them at once, or a init_with_config_or_else() type of function, where it tries to use config() but if it fails just sets default values. Is one of them alright?

Reply:
I personally dont like pulling the config() into the *Event in any way.